securedrop_client/api_jobs/updatestar.py

```diff
@@ -4,15 +4,14 @@
 from sdclientapi import API, RequestTimeoutError, ServerConnectionError
 from sqlalchemy.orm.session import Session
 
-from securedrop_client.api_jobs.base import ApiJob
+from securedrop_client.api_jobs.base import SingleObjectApiJob
 
 logger = logging.getLogger(__name__)
 
 
-class UpdateStarJob(ApiJob):
-    def __init__(self, source_uuid: str, is_starred: bool) -> None:
-        super().__init__()
-        self.source_uuid = source_uuid
+class UpdateStarJob(SingleObjectApiJob):
```

Comment:
Doesn't this change prevent any other `SingleObjectApiJob` that operates on the same source from entering the queue? What if there is a different type of job that derives from `SingleObjectApiJob` with the same source uuid in the queue?  To support this scenario we could check uuid **+ the type of job**.